frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -137,6 +137,18 @@ const FileHandler = ({
         validateOnly
     );
 
+    if ((reportId || errors.length || overallStatus) && !isWorking) {
```

Comment:
Ah!  So what I mean was something more this:

```ts
const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    // other stuff    
    dispatch({ type: FileHandlerActionType.PREPARE_FOR_REQUEST });
   

    let eventData;

    try {
        const response = await sendFile({
            contentType: contentType,
            fileContent: fileContent,
            fileName: fileName,
            client: client,
        });

        eventData = {
            warningCount: response?.warnings?.length,
            errorCount: response?.errors?.length,
        };
    } catch (e: any) {
        if (e.data) {
            eventData = {
                warningCount: e.data.warningCount,
                errorCount: e.data.errorCount,
            };
        }
    }

    if (eventData) {
        trackAppInsightEvent(EventName.FILE_VALIDATOR, {
            schema: sender?.schemaName,
            fileType: fileType,
            sender: organization?.name,
            ...eventData,
        });
    }
};
```

The reason for this is that it's better to invoke functions we don't intend to run on every render in an event handler (`onChange`, `onClick`, et cetera).  There are of course some exceptions to this, but this runs the risk of firing `trackAppInsightEvent` on every render when `(reportId || errors.length || overallStatus) && !isWorking)` is true, even if there wasn't a state change here.

Reply:
Thanks @stephenkao . This is great.